### PR TITLE
fix sql manager add

### PR DESCRIPTION
**Pull request by alex-taxiera**

### CODE REVIEW

**Review by exoRift — APPROVED** [review]

File: lib/sql-manager/index.js [hunk 1]
```diff
@@ -9,60 +9,55 @@ try {
...
-        table: object.type,
-        data: object._data,
-        where: { id: object.id }
-      })
-
-      return {
-        ...res
-      }
+      const [ res ] = await this._qb(object.type, '*')
+        .where({ id: object.id })
```
Comment (exoRift) [inline lib/sql-manager/index.js:41]: This is lexically backward. Your `where` should be after the update. Not to mention `where` is after the action in other instances

File: lib/sql-manager/index.js [hunk 2]
```diff
@@ -9,60 +9,55 @@ try {
...
     }
 
     async get (query) {
-      return this._qb.get({
-        table: query.type,
-        where: { [query.getKey]: query.getValue }
-      })
+      const rows = await this._qb(query.type)
+        .select('*')
+        .where({ [query.getKey]: query.getValue })
```
Comment (exoRift) [inline lib/sql-manager/index.js:50]: Don't construct an object for no reason. You can do `.where(query.getKey, query.getValue)`

File: lib/sql-manager/index.js [hunk 3]
```diff
@@ -9,60 +9,55 @@ try {
...
+
+      return rows[0]
     }
 
     async find (query) {
```
Comment (exoRift) [inline lib/sql-manager/index.js:55]: Maybe add a param to specify columns instead of just selecting all columns

File: lib/sql-manager/index.js [hunk 4]
```diff
@@ -9,60 +9,55 @@ try {
...
+
+      return res
     }
 
     async get (query) {
-      return this._qb.get({
-        table: query.type,
-        where: { [query.getKey]: query.getValue }
-      })
+      const rows = await this._qb(query.type)
```
Comment (exoRift) [inline lib/sql-manager/index.js:48]: Add a `.limit(1)`

File: lib/sql-manager/index.js (hunk 3 above)
Comment (exoRift) [reply lib/sql-manager/index.js:55]: Also, maybe object support instead of strictly where building?

File: lib/sql-manager/index.js [hunk 5]
```diff
@@ -9,60 +9,55 @@ try {
...
-        where: (builder) => this._whereBuilder(query, builder)
-      })
+      const rows = await this._qb(query.type)
+        .select('*')
+        .where((builder) => this._whereBuilder(query, builder))
+
+      return rows[0] ? rows : []
     }
 
     _whereBuilder (query, builder) {
```
Comment (exoRift) [inline lib/sql-manager/index.js:63]: This is very mushy code gravy. Beautify it and clean it up a bit.

File: lib/sql-manager/index.js [hunk 6]
```diff
@@ -117,6 +112,52 @@ try {
         default: throw Error('messed up query')
       }
     }
+
+    createTable ({ name, columns }) {
```
Comment (exoRift) [inline lib/sql-manager/index.js]: Switch the column object format to be
```js
{
  name: 'name',
  type: 'string',
  default: 'default',
  constraints: {
    maxLength: 255,
    nullable: false,
    primary: true,
    foreign: {
      references: ['table', 'column'],
      onUpdate: 'cascade',
      onDelete: 'cascade'
    }
}
```

File: lib/sql-manager/index.js [hunk 7]
```diff
@@ -117,6 +112,52 @@ try {
...
+      return this._qb.schema.hasTable(name)
```
Comment (exoRift) [inline lib/sql-manager/index.js]: Clean up this code gravy

File: lib/sql-manager/index.js [hunk 8]
```diff
@@ -117,6 +112,52 @@ try {
...
+            .then(() => {
+              return {
+                name,
+                columns
+              }
+            })
+        })
+    }
+
+    dropTable (table) {
```
Comment (exoRift) [inline lib/sql-manager/index.js]: If the table exists, return the table name. If it doesn't, return `undefined`

File: lib/sql-manager/index.test.js [hunk 9]
```diff
@@ -65,17 +66,20 @@ test.before(async (t) => {
   })
 
   for (const person of Object.values(t.context.people.data)) {
-    await dbm._qb.insert({
-      table: t.context.people.db,
-      data: person
-    })
+    await dbm._qb(t.context.people.db).insert(person)
```
Comment (exoRift) [inline lib/sql-manager/index.test.js:92]: Why not use `add`?

**Review by exoRift — CHANGES_REQUESTED** [review]
A lot of things were just lazily ported over rather than rewritten with our new standards. You of all people should want the final code to be clean and concise.

File: lib/sql-manager/index.js (hunk 3 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js:55]: we will always take all columns to supply the full object, there are no partials here

File: lib/sql-manager/index.js (hunk 5 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js:63]: explain

File: lib/sql-manager/index.js (hunk 6 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js]: this has been removed

File: lib/sql-manager/index.test.js (hunk 9 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.test.js:92]: didn't want the hook to depend on something that is being tested in this unit

File: lib/sql-manager/index.js (hunk 8 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js]: this has been removed

File: lib/sql-manager/index.js (hunk 4 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js:48]: done

File: lib/sql-manager/index.js (hunk 2 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js]: done

File: lib/sql-manager/index.js (hunk 1 above)
Comment (alex-taxiera) [reply lib/sql-manager/index.js]: done

File: lib/sql-manager/index.js (hunk 5 above)
Comment (exoRift) [reply lib/sql-manager/index.js:64]: It's extremely difficult to understand anything going on in there. It's very squished together and hard to follow

**Review by exoRift — APPROVED** [review]
Besides for the where builder needing a refactor, these are good.

Glad to see SK phased out
